app/models/course/assessment/question/text_response_comprehension_solution.rb

```diff
@@ -1,11 +1,13 @@
 # frozen_string_literal: true
+require 'rwordnet'
 class Course::Assessment::Question::TextResponseComprehensionSolution < ApplicationRecord
   self.table_name = 'course_assessment_question_text_response_compre_solutions'
 
   enum solution_type: [:compre_lifted_word, :compre_keyword]
 
   before_validation :remove_blank_solution,
                     :strip_whitespace_solution,
+                    :convert_solution_to_lemma
```

Comment:
Would suggest you merge the various methods together and rename as `sanitize_solution_and_derive_lemma` or something like that.

Reply:
And I think we should only invoke the methods if the solution changes, and if it doesn't, we shouldn't be invoking these callbacks.

Reply:
> Would suggest you merge the various methods together and rename as `sanitize_solution_and_derive_lemma` or something like that.

Okay.

> And I think we should only invoke the methods if the solution changes, and if it doesn't, we shouldn't be invoking these callbacks.

How should I 'check' if solution has been changed...?